src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/compute/v4_9/CloudStackComputePluginTest.java

```diff
@@ -1,894 +1,1426 @@
+        GetAllServiceOfferingsResponse.ServiceOffering serviceOffering =
+                Mockito.mock(GetAllServiceOfferingsResponse.ServiceOffering.class);
+        Mockito.doReturn(serviceOffering).when(this.plugin).getServiceOffering(
+                Mockito.eq(order) , Mockito.any(CloudStackUser.class));
 
-        String expectedServiceOfferingsRequestUrl = generateExpectedUrl(endpoint, serviceOfferingsCommand,
-                RESPONSE_KEY, JSON);
-        String expectedDiskOfferingsRequestUrl = generateExpectedUrl(endpoint, diskOfferingsCommand,
-                RESPONSE_KEY, JSON);
-        String serviceOfferingResponse = getListServiceOfferrings(fakeServiceOfferingId, "fake-service-offering",
-                Integer.parseInt(FAKE_CPU_NUMBER), Integer.parseInt(FAKE_MEMORY), FAKE_TAGS);
-        String diskOfferingResponse = getListDiskOfferrings(fakeDiskOfferingId, Integer.parseInt(FAKE_DISK),true);
+        GetAllDiskOfferingsResponse.DiskOffering diskOffering =
+                Mockito.mock(GetAllDiskOfferingsResponse.DiskOffering.class);
+        int orderDisk = order.getDisk();
+        Mockito.doReturn(diskOffering).when(this.plugin).getDiskOffering(
+                Mockito.eq(orderDisk), Mockito.any(CloudStackUser.class));
 
-        Mockito.when(this.client.doGetRequest(Mockito.eq(expectedServiceOfferingsRequestUrl), Mockito.eq(FAKE_TOKEN)))
-                .thenReturn(serviceOfferingResponse);
-        Mockito.when(this.client.doGetRequest(Mockito.eq(expectedDiskOfferingsRequestUrl), Mockito.eq(FAKE_TOKEN)))
-                .thenReturn(diskOfferingResponse);
+        String fakeUserDataString = "anystring";
+        Mockito.when(this.launchCommandGeneratorMock.createLaunchCommand(Mockito.any(ComputeOrder.class)))
+                .thenReturn(fakeUserDataString);
 
-        // exercise
-        ComputeOrder order = new ComputeOrder(null, FAKE_CLOUD_NAME, FAKE_MEMBER, FAKE_MEMBER, FAKE_INSTANCE_NAME,
-                Integer.parseInt(FAKE_CPU_NUMBER), Integer.parseInt(FAKE_MEMORY),
-                Integer.parseInt(FAKE_DISK), fakeImageId, userData, FAKE_PUBLIC_KEY, fakeNetworkdIds);
-        order.setRequirements(fakeRequirements);
+        DeployVirtualMachineResponse deployVirtualMachineResponse =
+                Mockito.mock(DeployVirtualMachineResponse.class);
+        Mockito.doReturn(deployVirtualMachineResponse).when(this.plugin)
+                .requestDeployVirtualMachine(Mockito.any(), Mockito.eq(cloudStackUser));
+
+        Mockito.doNothing().when(this.plugin).updateComputeOrder(
+                Mockito.eq(order), Mockito.eq(serviceOffering), Mockito.eq(diskOffering));
 
-        String createdVirtualMachineId = this.plugin.requestInstance(order, FAKE_TOKEN);
+        // exercise
+        this.plugin.requestInstance(order, cloudStackUser);
 
-        Mockito.verify(this.client, Mockito.times(1))
-                .doGetRequest(expectedServiceOfferingsRequestUrl, FAKE_TOKEN);
+        // verify
```

Comment:
you must verify if doRequestInstance() and the other aux methods have been called too